equivariant_attention/modules.py

```diff
@@ -342,6 +342,52 @@ def forward(self, features, **kwargs):
         return output
 
 
+#####################################
+class G1x1SE3_q(nn.Module):
+    """Graph Linear SE(3)-equivariant layer, equivalent to a 1x1 convolution.
+
+    This is equivalent to a self-interaction layer in TensorField Networks.
+    """
+    def __init__(self, f_in, f_out, fv_in, fv_out, learnable=True):
+        """SE(3)-equivariant 1x1 convolution.
+
+        Args:
+            f_in: input Fiber() of feature multiplicities and types
+            f_out: output Fiber() of feature multiplicities and types
+        """
+        super().__init__()
+        self.f_in = f_in
+        self.f_out = f_out
+        self.fv_in = fv_in
+        self.fv_out = fv_out
+
+        # Linear mappings: 1 per output feature type
+        self.transform = nn.ParameterDict()
+        for m_out, d_out in self.f_out.structure:
+            for mv_in, dv_in in self.fv_in.structure:
+                for mv_out, dv_out in self.fv_out.structure:
```

Comment:
Should we use `self.fv_in.degrees` and `self.fv_out.degrees`?

Reply:
we can change but is it necessary? they are similar to this except we dont need mv_in and mv_out right?